### PR TITLE
WIP: run against dbus-broker bench

### PR DESCRIPTION
**Pull request by elmarco**
This series allows to run busd against dbus-broker benchmark, with:
```
DBUS_BROKER_TEST_DAEMON=path-to-busd/target/debug/busd dbus-broker/build/test/dbus/bench-connect
```
The most important change is the introduction of `<busconfig>` parsing, and then handling of a few command line options.

The test currently hangs after what looks like an initial successful connection. Further investigation needed.

/!\ This branch needs https://github.com/dbus2/zbus/pull/390

### CODE REVIEW

**Comment by zeenix** [comment]
@elmarco the config changes are super desirable even w/o dbus-broker tests in mind. Do you think you can clean this up soon?

**Comment by zeenix** [comment]
@elmarco I rebased this branch on latest git main and put it [here](https://github.com/zeenix/busd/tree/dbus-broker-bench). I didn't push it to your branch because I had to remove the unmerged zbus API for it to build.

File: src/bin/busd.rs [hunk 1]
```diff
@@ -51,10 +66,30 @@ impl From<AuthMechanism> for zbus::AuthMechanism {
...
+    // FIXME: we don't support multiple <listen> atm
+    let address = config
+        .elements
+        .iter()
+        .rev()
+        .find_map(|e| match e {
+            config::Element::Listen(l) => Some(l),
+            _ => None,
+        })
+        .unwrap();
```
Comment (zeenix) [inline src/bin/busd.rs:112]: I think `BusConfig` should provide specific API for each element so user doesn't have to deal with generic elements. It's exposing the detail that config is xml-based IMO.

**Comment by jokeyrhyme** [comment]
@elmarco howdie, I mucked about today and came up with a slightly different take on the XML configuration file work over in #146 

There may or may not be something you like over there, but it's completely up to you :)

**Comment by elmarco** [comment]
@jokeyrhyme I have lost interest to help with this, I do not understand @zeenix project management.

**Comment by jokeyrhyme** [comment]
Oh, no worries, I can pick this up, hopefully this weekend :crossed_fingers:

**Pull request closed without merging**